snap/snapcraft.yaml

```diff
@@ -21,15 +21,17 @@ parts:
   connectedhomeip:
     plugin: nil
     build-environment:
-      - TAG: master
+      - UPSTREAM_TAG: master
+      - PACKAGE_VERSION: snap
```

Comment:
I suggest moving these to global variables, i.e. top level [`environment` dictionary](https://snapcraft.io/docs/snapcraft-yaml-reference), for better visibility.

Reply:
The global environment does not seem to be copied to the build environment. When I moved these variables to the global environment, they are not defined in the build environment. Relevant code: https://github.com/canonical/snapcraft/pull/2322/files#diff-42c4ab608a93b91e833534a2b90f71e2288196c3c3970cafd36648fdc952a6f3R256

Reply:
Oh yes, I didn't notice that these were build environments.